packages/pages-shared/asset-server/handler.ts

```diff
@@ -308,12 +308,14 @@ export async function generateHandler<
 
 			const clonedResponse = response.clone();
 
-			if (waitUntil) {
+			if (
+				waitUntil &&
+				response.headers.get("Content-Type")?.includes("text/html")
```

Comment:
Further down, we do `contentType.startsWith("text/html")`: https://github.com/cloudflare/wrangler2/blob/cbbc932f931c45b51ed735557e7b7ded9fbf1c34/packages/pages-shared/asset-server/handler.ts#L509

Should that be `contentType.includes("text/html")` like you have here? Maybe worth just extracting out to an `isHTML: (contentType: string) => boolean` so we're consistent in both places?

Reply:
Actually, I think it _should be_ `.startsWith(...)`. At the time of me writing the code, I wasn't sure if the type could be anywhere in the string, but it does seem that clients expect the type to come first, then directives. I'll abstract to a function and use `.startsWith(...)`